posthog/hogql/database/schema/query_log_archive.py

```diff
@@ -0,0 +1,227 @@
+            elif name == "status":
+                return ast.Alias(alias=name, expr=ast.Field(chain=[raw_table_name, "type"]))
+            elif name == "is_personal_api_key_request":
+                cmp_expr = ast.CompareOperation(
+                    op=ast.CompareOperationOp.Eq,
+                    left=ast.Constant(value="personal_api_key"),
+                    right=ast.Field(chain=[raw_table_name, "lc_access_method"]),
+                )
+                return ast.Alias(alias=name, expr=cmp_expr)
+            elif name == "cost_usd":
+                cost_cpu = ast.ArithmeticOperation(
+                    left=ast.Field(chain=[raw_table_name, "ProfileEvents_OSCPUVirtualTimeMicroseconds"]),
+                    right=ast.Constant(value=0.000000001),
+                    op=ast.ArithmeticOperationOp.Mult,
+                )
+                cost_bytes = ast.ArithmeticOperation(
+                    left=ast.Field(chain=[raw_table_name, "read_bytes"]),
+                    right=ast.Constant(value=0.000000000005),  # $5 per TB
+                    op=ast.ArithmeticOperationOp.Mult,
+                )
+                cost_ex = ast.ArithmeticOperation(
+                    left=cost_cpu,
+                    right=cost_bytes,
+                    op=ast.ArithmeticOperationOp.Add,
+                )
+                return ast.Alias(alias=name, expr=cost_ex)
+            else:
+                field = QUERY_LOG_ARCHIVE_FIELDS[name]
+                if field and isinstance(field, DatabaseField):
+                    return ast.Alias(alias=name, expr=ast.Field(chain=[raw_table_name, field.name]))
+                return ast.Alias(alias=name, expr=ast.Field(chain=[raw_table_name, *chain]))
+
+        fields: list[ast.Expr] = [get_alias(name, chain) for name, chain in requested_fields.items()]
+
+        return ast.SelectQuery(
+            select=fields,
+            select_from=ast.JoinExpr(table=ast.Field(chain=[raw_table_name])),
+            where=ast.And(
+                exprs=[
+                    ast.CompareOperation(
+                        op=ast.CompareOperationOp.Eq,
+                        left=ast.Constant(value=context.team_id),
+                        right=ast.Field(chain=["lc_team_id"]),
+                    ),
+                    ast.CompareOperation(
+                        op=ast.CompareOperationOp.Eq,
+                        left=ast.Constant(value="HogQLQuery"),
+                        right=ast.Field(chain=["lc_query__kind"]),
+                    ),
+                ]
+            ),
+        )
+
+
+class RawQueryLogArchiveTable(FunctionCallTable):
+    fields: dict[str, FieldOrTable] = RAW_QUERY_LOG_ARCHIVE_FIELDS
+
+    name: str = "raw_query_log_archive"
+
+    def to_printed_clickhouse(self, context) -> str:
+        return "query_log_archive"
+
+    def to_printed_hogql(self) -> str:
+        return "query_log"
```

Comment:
We have no guard on making sure people dont query other teams queries with this (I think) - is that right?

Reply:
this is true, although this table is not querable